src/compiler/abstract_compiler.nit

```diff
@@ -1377,6 +1391,12 @@ abstract class AbstractCompilerVisitor
 	# The method is unsafe and is just a direct wrapper for the specific implementation of native arrays
 	fun native_array_set(native_array: RuntimeVariable, index: Int, value: RuntimeVariable) is abstract
 
+        # Instantiate a new routine pointer
+        fun routine_ref_instance(routine_mclass_type: MClassType, recv: RuntimeVariable, mmethoddef: MMethodDef): RuntimeVariable is abstract
```

Comment:
Shouldn't you give the whole callsite instead of a single mmethoddef ?

Reply:
If you remember at the beginning we figured out that the call site information will be lost during the actual call. Thus, we only need mmethoddef for creating an instance.